docs/0___preamble.adoc

```diff
@@ -93,68 +93,65 @@ This document defines the Functional Mock-up Interface (FMI),
-a dynamic system model that can be utilized by other modeling and simulation environments.
-Models are described by differential,
-algebraic and discrete equations with time-, state- and step-events.
-If the C code describes a continuous system,
-this system is solved with the integrators of the environment where it is used.
-The models to be treated by this interface can be large for usage in offline or online simulation,
-or they can be used in embedded control systems on microprocessors.
-
 _FMI for Co-Simulation (<<fmi-for-co-simulation>>)_
 
 The intention is to provide an interface standard for coupling
-of simulation tools in a co-simulation environment.
+of simulation models or tools in a co-simulation environment.
 The data exchange between subsystems is restricted to discrete communication points.
 In the time between two communication points,
-the subsystems are solved independently from each other by their individual solver.
-Master algorithms control the data exchange between subsystems and the
-synchronization of all simulation solvers (slaves).
-Simple master algorithms,
-as well as more sophisticated ones are both supported.
+the subsystems are solved independently by internal means, see below.
+Master algorithms control the data exchange and the
+synchronization between subsystems (FMUs).
+
 Note that the master algorithm itself is not part of the FMI standard.
 
+_FMI for Hybrid Co-Simulation_
+
+The Hybrid Co-Simulation interface adds a number of features primarily to allow for more sophisticated master algorithms that aim at more efficient and robust simulations.
+Such additional features are raising events between communication time points using synchronous and asynchronous clocks or sharing values between communication points to allow for improved interpolation of data.
+
+For both Co-Simulation and Hybrid Co-Simulation the master algorithm is shielded from how the subsystem FMU advances time internally.
+For example, FMUs containing ODEs and exposing either of the Co-Simulation interfaces require to include an ODE solver inside the FMU to internally advance time between the communication points.
+Or, for controller code packaged as FMU, an internal scheduling algorithm will trigger tasks at the correct time and order while advancing time to the next communication point or event.
+
+_FMI for Model Exchange (<<fmi-for-model-exchange>>)_
+
+The Model Exchange interface exposes an ODE to an external solver of an importing tool's master algorithm.
+Models are described by differential, algebraic and discrete equations with time-, state- and step-events.
+That master algorithm, usually an ODE or DAE solver, like DASSL, is responsible to advance time, set states and handle events etc.
+
+_FMI for Scheduled Execution Simulation_
+
+The Scheduled Execution Simulation interface exposes individual parts or tasks of a control algorithm to be called from an external scheduler of an importing tool's master algorithm.
```

Comment:
...individual parts of a model or tasks of a control algorithm...

because not al models are control algorithms in this mode.